File: 125.R
# Palindromic sums

Args  <- commandArgs()
limit <- ifelse(is.na(Args[6]), 1e8, as.numeric(Args[6]))

is_palindrome <- function(number) {
    digit_chars <- strsplit(as.character(as.integer(number)), split=NULL)[[1]]
    reversed_number <- as.integer(paste(rev(digit_chars), collapse=''))
    if ( number == reversed_number ) {
        return(TRUE)
    } else {
        return(FALSE)
    }
}

squares <- 1
n <- 1
while ( squares[length(squares)] < limit ) {
    n <- n + 1
    squares <- c(squares, n * n)
}

palindromes <- vector()

for ( begin in seq.int(length(squares) - 1) ) {
    square_sum <- squares[begin]
    n <- begin + 1
    while ( n < length(squares) ) {
        square_sum <- square_sum + squares[n]
        if ( square_sum > limit ) {
            break
        }
        if ( is_palindrome(square_sum) ) {
            palindromes <- c(palindromes, square_sum)
        }
        n <- n + 1
    }
}

cat(sum(unique(palindromes)), fill=TRUE)

File: 125.js
#!/usr/bin/env node

// Palindromic sums

var program = require('commander');

program
    .version('0.1.0')
    .description('Palindromic sums')
    .option('-l, --limit <int>', 'The maximum number', Number, 1e8)
    .parse(process.argv);

var squares = [1];
var n = 1;
while (squares[squares.length - 1] < program.limit) {
    n++;
    squares.push(n * n);
}

var palindrome = {};

for (var start = 1; start <= squares.length - 1; start++) {
    var square_sum = squares[start - 1];
    var next = start + 1;
    while (next < squares.length) {
        square_sum += squares[next - 1];
        if (square_sum > program.limit) {
            break;
        }
        if (is_palindrome(square_sum)) {
            palindrome[square_sum] = true;
        }
        next++;
    }
}

var total = 0;
for (num in palindrome) {
    total += parseInt(num);
}

console.log(total);

function is_palindrome(number) {
    return number.toString().split('').reverse('').join('') == number;
}
